pages/documents/DeveloperTools/FaaS/Developing/best-pratices.md

```diff
@@ -0,0 +1,241 @@
+
+### Error handling in async functions
+
+If an async call has no correct error handling implemented the function will not end in the execution time and the error `Lambda Execution is taking too long` will be thrown.
+
+For [then-catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) promises it is necessary to have a `catch` at the end of a `then`.
+
+```javascript
+const myPromise =
+  (new Promise(myExecutorFunc))
+  .then(handleFulfilledA)
+  .then(handleFulfilledB)
+  .then(handleFulfilledC)
+  .catch(handleRejectedAny);
+```
+
+For [async-await](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Operators/await) a try-catch is necessary. 
+
+```javascript
+try {
+  await myPromise();
+} catch(error) {
+  // handle error
+} 
+```
+
+### Async/Await Functions
+
+It is possible to use the [async/await](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Operators/await) functionality within the LivePerson Functions. It's important to wrap the await in a try/catch, otherwise the function will have some weird behavior and errors will not be catched and displayed correctly.
+
+The example shows the HTTP-Template with async/await.
+
+```javascript
+async function lambda(input, callback) {
+  // Importing the FaaS Toolbelt
+  const { Toolbelt } = require("lp-faas-toolbelt");
+  // Obtain an HTTPClient Instance from the Toolbelt
+  const httpClient = Toolbelt.HTTPClient(); // For API Docs look @ https://www.npmjs.com/package/request-promise
+
+  const URL = "https://www.liveperson.com";
+
+  try {
+    const response = await httpClient(URL, {
+      method: "GET", // HTTP VERB
+      headers: {}, // Your Headers
+      simple: false, // IF true => Status Code != 2xx & 3xx will throw
+      resolveWithFullResponse: true //IF true => Includes Status Code, Headers etc.
+    });
+      
+    const { statusCode, body } = response;
+    // Perform Action based on Status Code
+    switch (statusCode) {
+      case 200: 
+        return callback(null, statusCode);
+      // If not Whitelisted Proxy will reject with 403. Body contains also a message indicating that
+      case 403:
+        return callback(new Error("Domain is not whitelisted"), null);
+      default:
+        return callback(
+          new Error(`Recieved unexpected status code ${statusCode}`)
+        );
+    }
+  } catch (err) {
+    callback(err, null)
+  }
+}
+```
+
+### Limitations of Timeouts
+
+The usage of `setTimeout()` in LivePerson Functions must be handled with care. 
+* Functions cannot exceed the runtime of **30 secconds**. The error `Lambda Execution is taking too long` will thrown.
+* If you execute multiple functions with a `setTimeout()` implemented, the event loop of the service will fill up and even functions with a lower execution time than 30 seconds will take longer and may reach the max. execution time.
```

Comment:
```suggestion
* If you execute multiple functions containing a `setTimeout()` call, the event loop of the service may fill up. Even functions with a lower execution time than 30 seconds could therefore take longer, may reach the execution time limit and fail. Because of this timeouts in general should be handled with care and it is encouraged to double check their need in any use case.
```